src/Native/System.Net.Security.Native/pal_gssapi.h

```diff
@@ -0,0 +1,141 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#if HAVE_GSSFW_HEADERS
+typedef struct gss_name_t_desc_struct GssName;
```

Comment:
Actually we are using two different implementations of the [same standard native API](https://tools.ietf.org/html/rfc2743).  On OSX, we are using the built-in heimdal implementation. On linux, we are using mit implementation of the same api.
  The standard api uses names like ```gss_name_t``` , ```gss_buffer_t```.  The implementations differ in terms of what these types actually mean.
 On linux (with mit implementation of gssapi), ```gss_name_t``` refers to ```struct gss_name_struct *```. 
But on OSX (with heimdal implementation) ```gss_name_t``` means ```struct gss_name_t_desc_struct *```.

Based on your earlier comment, in order for me to have forward declarations of the native types that are used in this header file, I need to forward declare the underlying concrete implementation as well as the types that are being used. Since the names of concrete implementations were different, I introduced a new set of typedefs. That way, am able to localize the implementation-specific differences to just one place.  

The alternative is to introduce  ```#if HAVE_GSSFW_HEADERS``` through out the header file and the cpp file wherever the objects are being referred to.
  Or to use two different versions of  the files. 
 Please let me know. I will change it the way you prefer.